### ISSUE TITLE
Configure both element and selector together ...

### ISSUE THREAD

**Issue by ColinEberhardt** [d3fc/d3fc-data-join#1]
Replace:

```
var join = dataJoin()
            .element('li')
            .selector('li');
```

with 

```
var join = dataJoin('li')
```

And replace

```
var join = dataJoin()
            .element('li')
            .selector('li.animal')
            .attr('class', 'animal');
```

With:

```
var join = dataJoin('li.animal');
```

This would simplify the vast majority of uses in d3fc.

We've got a couple of options here:

1. The dataJoin('li.animal') form could return the regular data-join component, which would mean further calls to `element` and `selector` could bemused to create a 'broken' join.
2. Use specific functions, `classedJoin('li', 'animal')` and `elementJoin('li')`, which do not expose any of the dataJoin properties (other than `key`)

**Comment by chrisprice** [comment]
I've noticed a pattern in the new [d3-force forces](https://github.com/d3/d3-force/blob/master/README.md#forceCollide) whereby something similar to 1 is used. I think I'd prefer to see a mix of the two ideas though -
```js
dataJoin('li', 'animal') // returns a dataJoin with element, selector and attr pre-configured.
````

**Comment by chrisprice** [comment]
Fixed by #6
